server/src/main/java/org/elasticsearch/index/mapper/NestedValueFetcher.java

```diff
@@ -50,7 +50,7 @@ public List<Object> fetchValues(SourceLookup lookup, List<Object> includedValues
         for (Object entry : nestedValues) {
             // add this one entry only to the stub and use this as source lookup
             stub.put(nestedFieldName, entry);
-            SourceLookup nestedSourceLookup = new SourceLookup();
+            SourceLookup nestedSourceLookup = new SourceLookup.Static();
             nestedSourceLookup.setSource(filteredSource);
```

Comment:
Seeing this makes me think "SourceLookup.Static should take the source as a parameter - but I think that'd only work sometimes, right? In the fetch phase we don't have it available right away, I think.